### PR TITLE
Refactor CI script

### PR DESCRIPTION
**Pull request by ichizok**
I think should do all tests in order to expose all existing problems except when configure/build fail.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -39,33 +39,37 @@ env:
...
+  - grep -q -- "-DDYNAMIC_PERL_DLL=\\\\\"${vi_cv_dll_name_perl}\\\\\"" src/auto/config.mk
+  - grep -q -- "-DDYNAMIC_PYTHON_DLL=\\\\\"${vi_cv_dll_name_python}\\\\\"" src/auto/config.mk
+  - grep -q -- "-DDYNAMIC_PYTHON3_DLL=\\\\\"${vi_cv_dll_name_python3}\\\\\"" src/auto/config.mk
+  - echo -en "travis_fold:end:configure\\r\\033[0K"
+  - echo -e "\\033[33;1mBuilding MacVim\\033[0m" && echo -en "travis_fold:start:build\\r\\033[0K"
+  - make -j${NPROC}
+  - echo -en "travis_fold:end:build\\r\\033[0K"
+  - set +o errexit
+  - echo -e "\\033[33;1mTesting MacVim\\033[0m" && echo -en "travis_fold:start:test\\r\\033[0K"
+  - macvim_excmd() { ${VIMCMD} -u NONE -i NONE -g -f -X -V1 -es "$@" -c 'echo ""' -c quit 2>&1; }
```
Comment (ychin) [inline .travis.yml]: Why remove the `redit>result` and grep in the end?

**Comment by ychin** [comment]
Let me think about this a little bit. I'm a little hesitant because I find Travis CI to have too long of a turnaround time these days (30+ minutes) and a small issue can drag on forever and slow down iteration. While exposing more errors is good, I find that halting immediately tend to give a fast feedback to the user and when you scroll down to the end of the log that's where the error is instead of needing to scroll up the massive amount of log spam from successful tests and search around.

**Comment by ychin** [comment]
Hi @ichizok is this still work in progress? I noticed you renamed the title to "[WIP]" a while ago but wondering if there are still changes that need to go in.

File: .travis.yml (hunk 1 above)
Comment (ichizok) [reply .travis.yml]: I think these one-line tests should check whether each of if_xxx commands work properly or not (minimal test), and output should be tested in `make test`.

**Comment by ichizok** [comment]
There are the below major factors causing tests take long time:

### `brew update`

In all envs. This can resolve to use directory cache.

### Upgrading brew packages

Especially in xcode7.3: packages are built from their sources, not using bottles.
Thus I think should enable the language interfaces (if_xxx) only in xcode10.2, or at least disable in xcode7.3.

### `Test_writedelay` (test_options.vim) on GUI

Except in xcode7.3.

https://travis-ci.org/macvim-dev/macvim/builds/555492863

xcode7.3:
```
Executed Test_writedelay() in   1.871324 seconds
```

xcode 8.3:
```
Executed Test_writedelay() in 205.001438 seconds
```

xcode9.4:
```
Executed Test_writedelay() in 272.267539 seconds
```

xcode10.2
```
Executed Test_writedelay() in 455.950352 seconds
```

These are caused by the behavior (bug?) of `set fullscreen`.
In test_options.vim `Test_set_values` does `set fullscreen`, `set nofullscreen`, and `set fullscreen&` but actual screen remains full-screen after first `set fullscreen` therefore redrawing screen spends long time when with `set writedelay`.

**Comment by ichizok** [comment]
Thank you, please merge.

**Pull request merged**